sphinx/texinputs/sphinxlatexobjects.sty

```diff
@@ -146,6 +146,21 @@
   \item[{#1\sphinxcode{(}\py@sigparams{#2}{#3}\strut}]
   \pysigadjustitemsep
 }
+\newcommand{\pysigwithmultilineargsret}[3]{%
+  % render each argument on its own line
+  \item[#1\sphinxcode{(}\strut]
+
+  \@for\param:=#2\do{
+    \param\sphinxcode{,}
+
+  }
+  \itemsep\z@skip
+  \item[\sphinxcode{)}#3\strut]
+  % alternative to above two lines, which aligns the closing parenthesis
+  % with the arguments:
+  % \noindent\sphinxcode{)}{#3}
```

Comment:
Now it is matter of taste if an isolated `)` at top of page before object description is or not a bad thing.  As far as I understand regarding the location of `)` either at start of line or underneath first character of last argument the [PEP8](https://peps.python.org/pep-0008/) considers both OK. I prefer the latter, which would be the LaTeX output if the above suggested change is applied (which also due to the `\nopagebreak` prevents a page break; in contrast to a `\nopagebreak` before an `\item` which does not work).

Reply:
One could also put the `)` on same line as last parameter.  However we then would like to add a `\nopagebreak` before that last parameter line.  But LaTeX has no simple variant of its `\@for` macro to loop *and* know when we are handling the last item.  This is only provided by some external rather extensive frameworks, or we must assume we have a recent LaTeX and can then tap into its "LaTeX3" programming layer, which however would make maintenance more complex to future maintainers as it raises the level of  LaTeX knowledge needed.

Reply:
Hmmmm I don' t know, I initially put the closing parenthesis dedented (in LaTeX but also in other formats) to follow Black's style. To quote from its [doc:](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#how-black-wraps-lines) "Also, having the closing bracket dedented provides a clear delimiter between two distinct sections of the code that otherwise share the same indentation level". So their argument is that by dedenting the parenthesis you see more clearly the start of the docstring. But well in documentation where most of the time the arguments and the docstring have a different style anyway, I don't know how relevant that is... And then there's this whole caveat about avoiding page breaks that you mention. So personally I don't mind the tradeoff of accepting your first suggestion above. If anyone has a strong opinion on this, let us know.

Reply:
> If anyone has a strong opinion on this, let us know.

I don't have a strong opinion but a slight preference for the closing brace aligned with first characters of arguments...

It is feasible to have it dedentet and at same time to prevent a page break but I have to plunge a bit into bowels of LaTeX's lists, o refresh my memory about what are the right dimensional parameters to use here (will do this).

Reply:
```suggestion
  % fulllineitems sets \labelwidth to be like \leftmargin
  \nopagebreak\noindent\kern-\labelwidth\sphinxcode{)}{#3}
```

This dedents the `)`, and prevents a page break right above it.  As the `)` is alone on its line, and somewhat to the left, the vertical space to object description looks a bit large but it is exactly the same as with `single-line-signature`, as one can check experimenting with using for example `)xxx....xxxx` in the LaTeX code.  Also tested if object description starts itself by a list.

Reply:
> the vertical space to object description looks a bit large

But this effect goes away with presence of type annotation.  Here is an example with our own documentation (the `python_maximum_signature_line_length` was set to very small to force one line per argument rendering)
![Capture d’écran 2023-01-24 à 11 34 18](https://user-images.githubusercontent.com/2589111/214269898-8f83862d-f6c6-4ead-a107-59f86e0d45f4.png)

(the annoying footnote markers in PDF are explained by the ``latex_show_urls = 'footnote'``)

Reply:
another example showing how very long type annotations wrap
![Capture d’écran 2023-01-24 à 11 40 47](https://user-images.githubusercontent.com/2589111/214271107-61d9dd7f-a9ee-4ac8-9dc0-07f083276fe6.png)

I forgot to say that parameters are left-aligned with description. 

I am fine with applying this which keeps in sync LaTeX and HTLM rendering.  The above suggested change of not using `\item`  for the closing `)` avoids a page break.

Regarding LaTeX, this PR looks good to me.

If we merge soon, the doc ``versionadded`` directive arguments in ``doc/usage/configuration.rst`` should be ``6.2.0``, not ``6.x``.

Reply:
Example with nesting:
![Capture d’écran 2023-01-24 à 11 55 41](https://user-images.githubusercontent.com/2589111/214273961-99c1cab8-98b9-4c56-8ea4-eb3f136d12ab.png)
Due to the way LaTeX is configured by default for nested lists, the indent from left compared to object name decreases the deeper we are in a sub-list.

Reply:
That's a great solution, thanks a lot for digging into that!

(It's always good to be reminded of the extent of your ignorance hahaha)

Reply:
@TLouf thanks to stimulation from your PR, I have updated master at 55170d01d to replace a hard-coded `\emph` by `\sphinxparam`.  This means once your PR is merged, it is even possible (to LaTeX geeks) to redefine `\sphinxparam` to do extra customization such as shifting horizontally the parameter on its line (styling inside is done by `\DUrole` mark-up).  But... as was mentioned elsewhere many projects do not bother about building the PDF docs... anyway let's be thorough... I like the way our own Sphinx doc renders to PDF with your PR and all signatures rendered one param per line.  The presence of type annotations avoids this visual artefact of increased vertical distance which I experience when a single `)` is all the way to the left on its own.